backends/tfhe-cuda-backend/cuda/src/integer/vector_find.cuh

```diff
@@ -0,0 +1,300 @@
+  }
+
+  copy_radix_ciphertext_slice_async<Torus>(
+      streams.stream(0), streams.gpu_index(0), temp_aggregated_vector, 0,
+      num_packed_blocks, aggregated_vector, 0, num_packed_blocks);
+
+  integer_radix_apply_univariate_lookup_table<Torus>(
+      streams, carry_ct, temp_aggregated_vector, bsks, ksks, carry_extract_lut,
+      num_packed_blocks);
+
+  integer_radix_apply_univariate_lookup_table<Torus>(
+      streams, message_ct, temp_aggregated_vector, bsks, ksks,
+      message_extract_lut, num_packed_blocks);
+
+  for (uint32_t index = 0; index < num_packed_blocks; index++) {
+    copy_radix_ciphertext_slice_async<Torus>(
+        streams.stream(0), streams.gpu_index(0), lwe_array_out, 2 * index,
+        2 * index + 1, message_ct, index, index + 1);
+
+    if (2 * index + 1 < lwe_array_out->num_radix_blocks) {
+      copy_radix_ciphertext_slice_async<Torus>(
+          streams.stream(0), streams.gpu_index(0), lwe_array_out, 2 * index + 1,
+          2 * index + 2, carry_ct, index, index + 1);
+    }
+  }
+
+  cuda_synchronize_stream(streams.stream(0), streams.gpu_index(0));
+}
+
+template <typename Torus>
+uint64_t scratch_cuda_aggregate_one_hot_vector(
+    CudaStreams streams, int_aggregate_one_hot_buffer<Torus> **mem_ptr,
+    int_radix_params params, uint32_t num_blocks, bool allocate_gpu_memory) {
+
+  uint64_t size_tracker = 0;
+  *mem_ptr = new int_aggregate_one_hot_buffer<Torus>(
+      streams, params, num_blocks, allocate_gpu_memory, size_tracker);
+
+  return size_tracker;
+}
+
+template <typename Torus>
+__host__ void host_unchecked_match_value(
+    CudaStreams streams, CudaRadixCiphertextFFI *lwe_array_out_result,
+    CudaRadixCiphertextFFI *lwe_array_out_boolean,
+    CudaRadixCiphertextFFI const *lwe_array_in_ct,
+    const uint64_t *h_match_inputs, const uint64_t *h_match_outputs,
+    int_unchecked_match_buffer<Torus> *mem_ptr, void *const *bsks,
+    Torus *const *ksks) {
+  host_compute_equality_selectors<Torus>(
+      streams, mem_ptr->selectors_list, lwe_array_in_ct,
+      mem_ptr->num_input_blocks, h_match_inputs, mem_ptr->eq_selectors_buffer,
+      bsks, ksks);
+
+  for (uint32_t i = 0; i < mem_ptr->num_matches; i++) {
+    copy_radix_ciphertext_slice_async<Torus>(
+        streams.stream(0), streams.gpu_index(0), mem_ptr->packed_selectors_ct,
+        i, i + 1, &mem_ptr->selectors_list[i], 0, 1);
+  }
+
+  if (!mem_ptr->max_output_is_zero) {
+    host_create_possible_results<Torus>(
+        streams, mem_ptr->possible_results_list, mem_ptr->selectors_list,
+        mem_ptr->num_matches, h_match_outputs, mem_ptr->possible_results_buffer,
+        bsks, ksks);
+  }
+
+  if (mem_ptr->max_output_is_zero) {
+    host_integer_is_at_least_one_comparisons_block_true<Torus>(
+        streams, lwe_array_out_boolean, mem_ptr->packed_selectors_ct,
+        mem_ptr->at_least_one_true_buffer, bsks, (Torus **)ksks,
+        mem_ptr->num_matches);
+    return;
+  }
+
+  host_aggregate_one_hot_vector<Torus>(
```

Comment:
If max_output_is_zero you could avoid allocating memory for  aggregate_one_hot_vector and is_at_least_one_comparisons_block_true